### ISSUE TITLE
Unable to open AI playground.. It hangs at loading screen.

### ISSUE THREAD

**Issue by jaact** [intel/AI-Playground#81]
## Describe the bug
Unable to open AI playground.. It hang at loading screen.
Tried installed the Latest Microsoft Visual C++ Redistributable Version - The latest version is 14.40.33816.0 
No Python installed in the Computer.

## To Reproduce
1. Installed AI playground
2. Launched AI Playground
3. Keep loading forever. (more than 30mins-1hr)

## Expected behavior
It should go into the AI Playground

## Environment (please complete the following information):

- OS: Windows11
- GPU: Intel Arc A750 8G
- CPU: 12 Gen Intel CPU
- Version: AI-Playground-v1.01b-dGPU

**Comment by Nuullll** [comment]
Per offline discussion with @jaact:

The error log (by following instruction [here](https://github.com/intel/AI-Playground/issues/57#issuecomment-2398761243)):
```
C:\Program Files\AI Playground\resources\service>..\env\python.exe .\web_api.py
C:\Program Files\AI Playground\resources\env\Lib\site-packages\torchvision\io\image.py:13: UserWarning: Failed to load image Python extension: 'Could not find module 'C:\Program Files\AI Playground\resources\env\Lib\site-packages\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.'If you don't plan on using image functionality from torchvision.io, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have libjpeg or libpng installed before building torchvision from source?
  warn(
ZE_LOADER_DEBUG_TRACE:Using Loader Library Path:
ZE_LOADER_DEBUG_TRACE:Tracing Layer Library Path: ze_tracing_layer.dll
2024-10-22 23:00:04,359 - INFO - intel_extension_for_pytorch auto imported
Traceback (most recent call last):
  File "C:\Program Files\AI Playground\resources\service\web_api.py", line 6, in <module>
    from sd_adapter import SD_SSE_Adapter
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\ipex_llm\utils\ipex_importer.py", line 76, in custom_ipex_import
    return RAW_IMPORT(name, globals, locals, fromlist, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\AI Playground\resources\service\sd_adapter.py", line 7, in <module>
    import paint_biz
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\ipex_llm\utils\ipex_importer.py", line 80, in custom_ipex_import
    return RAW_IMPORT(name, globals, locals, fromlist, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\AI Playground\resources\service\paint_biz.py", line 10, in <module>
    from diffusers import (
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\ipex_llm\utils\ipex_importer.py", line 76, in custom_ipex_import
    return RAW_IMPORT(name, globals, locals, fromlist, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\diffusers\_init_.py", line 5, in <module>
    from .utils import (
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\ipex_llm\utils\ipex_importer.py", line 76, in custom_ipex_import
    return RAW_IMPORT(name, globals, locals, fromlist, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\diffusers\utils\_init_.py", line 38, in <module>
    from .dynamic_modules_utils import get_class_from_dynamic_module
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\ipex_llm\utils\ipex_importer.py", line 76, in custom_ipex_import
    return RAW_IMPORT(name, globals, locals, fromlist, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\AI Playground\resources\env\Lib\site-packages\diffusers\utils\dynamic_modules_utils.py", line 28, in <module>
    from huggingface_hub import cached_download, hf_hub_download, model_info
ImportError: cannot import name 'cached_download' from 'huggingface_hub' (C:\Program Files\AI Playground\resources\env\Lib\site-packages\huggingface_hub\_init_.py)
```

which is related to huggingface_hub version issue (similar issue https://github.com/fudan-generative-vision/hallo2/issues/11)

```
C:\Program Files\AI Playground\resources\service>..\env\python.exe -m pip list
Package                     Version
--------------------------- ----------------------
absl-py                     2.1.0
accelerate                  1.0.1
addict                      2.4.0
aiohappyeyeballs            2.4.3
aiohttp                     3.10.10
aiosignal                   1.3.1
annotated-types             0.7.0
anyio                       4.6.2.post1
attrs                       24.2.0
backoff                     2.2.1
basicsr                     1.4.2
beautifulsoup4              4.12.3
bigdl-core-xe-21            2.5.0b20240715
bigdl-core-xe-addons-21     2.5.0b20240715
bigdl-core-xe-batch-21      2.5.0b20240715
blinker                     1.8.2
certifi                     2024.8.30
cffi                        1.17.1
chardet                     5.2.0
charset-normalizer          3.4.0
click                       8.1.7
colorama                    0.4.6
compel                      2.0.2
cryptography                43.0.3
dataclasses-json            0.6.7
datasets                    2.18.0
diffusers                   0.27.2
dill                        0.3.8
docstring_parser            0.16
docx2txt                    0.8
emoji                       2.14.0
eval_type_backport          0.2.0
faiss-cpu                   1.8.0
filelock                    3.16.1
filetype                    1.2.0
Flask                       3.0.3
frozenlist                  1.4.1
fsspec                      2024.2.0
future                      1.0.0
greenlet                    3.1.1
grpcio                      1.67.0
h11                         0.14.0
httpcore                    1.0.6
httpx                       0.27.2
huggingface-hub             0.26.1
idna                        3.10
imageio                     2.36.0
importlib_metadata          8.5.0
intel_extension_for_pytorch 2.1.40+xpu
ipex-llm                    2.1.0b20240715
itsdangerous                2.2.0
Jinja2                      3.1.4
joblib                      1.4.2
jsonpatch                   1.33
jsonpath-python             1.0.6
jsonpointer                 3.0.0
langchain                   0.2.11
langchain-community         0.2.10
langchain-core              0.2.23
langchain-text-splitters    0.2.4
langdetect                  1.0.9
langsmith                   0.1.136
lazy_loader                 0.4
lmdb                        1.5.1
lxml                        5.3.0
Markdown                    3.7
markdown-it-py              3.0.0
MarkupSafe                  3.0.2
marshmallow                 3.23.0
mdurl                       0.1.2
mpmath                      1.3.0
multidict                   6.1.0
multiprocess                0.70.16
mypy-extensions             1.0.0
nest-asyncio                1.6.0
networkx                    3.4.2
nltk                        3.9.1
numpy                       1.26.4
opencv-python               4.9.0.80
orjson                      3.10.9
packaging                   24.1
pandas                      2.2.3
peft                        0.10.0
pillow                      10.3.0
pip                         24.1.2
platformdirs                4.3.6
propcache                   0.2.0
protobuf                    5.28.2
psutil                      5.9.8
pyarrow                     17.0.0
pyarrow-hotfix              0.6
pycparser                   2.22
pydantic                    2.9.2
pydantic_core               2.23.4
Pygments                    2.18.0
pyparsing                   3.2.0
pypdf                       4.2.0
python-dateutil             2.8.2
python-iso639               2024.4.27
python-magic                0.4.27
pytz                        2024.2
PyYAML                      6.0.2
RapidFuzz                   3.10.0
regex                       2024.9.11
requests                    2.32.3
requests-toolbelt           1.0.0
rich                        13.9.2
safetensors                 0.4.5
scikit-image                0.24.0
scikit-learn                1.5.2
scipy                       1.14.1
sentence-transformers       2.3.1
sentencepiece               0.2.0
setuptools                  69.5.1
shtab                       1.7.1
six                         1.16.0
sniffio                     1.3.1
soupsieve                   2.6
SQLAlchemy                  2.0.36
sympy                       1.13.3
tabulate                    0.9.0
tb-nightly                  2.19.0a20241021
tenacity                    8.5.0
tensorboard-data-server     0.7.2
threadpoolctl               3.5.0
tifffile                    2024.9.20
tokenizers                  0.15.2
tomli                       2.0.2
torch                       2.1.0.post3+cxx11.abi
torchvision                 0.16.0.post3+cxx11.abi
tqdm                        4.66.5
transformers                4.39.0
trl                         0.9.3
typing_extensions           4.12.2
typing-inspect              0.9.0
tyro                        0.8.13
tzdata                      2024.2
unstructured                0.14.6
unstructured-client         0.26.1
urllib3                     2.2.3
Werkzeug                    3.0.4
wheel                       0.43.0
wrapt                       1.16.0
xxhash                      3.5.0
yapf                        0.40.2
yarl                        1.16.0
zipp                        3.20.2
```

The issue could be worked around by executing `..\env\python.exe -m pip install huggingface_hub==0.25.00`.

I'll check why this happened.

**Comment by jaact** [comment]
The workaround is working for me.
Thanks a lot for the speedy responds.

**Comment by ChuckTerry** [comment]
I had the same issue earlier today and can confirm the workaround works for me as well.

**Comment by Peter226** [comment]
Same here, workaround works perfectly.

**Comment by Undervolted-c** [comment]
Works with this workaround. 

_Thanks_.

**Comment by RogerWeihrauch** [comment]
Hi all

Same issue here - tried workaround - did not help here - setup of 'AI Playground' as 'only for me' - could be the issue here?

So, mentioned workaround does not work/help here; also added path to 'python.exe' as mentioned when installing huggingface: no success.

I set it up as 'only for me' and not for all Users so that the whole Ai Playground env will now be found in:
Target:
"C:\Users\RogiWanKenobi\AppData\Local\Programs\AI Playground\AI Playground.exe"
.. to be executed in:
"C:\Users\RogiWanKenobi\AppData\Local\Programs\AI Playground"

So, what do you think: could this be the issue on my side?

I am not very used to Windows, only more used to linux.

Thanks to all for any help; will be highly appreciated.
Kind regards,
Roger

**Comment by Peter226** [comment]
Do not manually install anything, just run the command Nuullll commented in the terminal / command line from the right path. I don't know why you added that python to windows path.
Also make sure you have a recent Intel driver on all of your Intel GPUs, integrated and discrete GPUs too. For me windows randomly downgrades my driver versions sometimes and it causes things like AI Playground and Blender to stop working properly even if it's just my integrated GPU that has an old driver.

**Comment by RogerWeihrauch** [comment]
@Peter226 
Well, added it to PATH because hint in CMD window when installing huggingface via pip did mention todo so.
Did install huggingface manually because was mentioned as workaround to get the AI Playground started.
So, what would you suggest here?
Drivers are on latest state.
BTW, I did install 'AI.Playground-v1.01b-Desktop_dGPU' since I only have an intel ARC A770 GPU and NOT an intel CPU.
Regards,
Roger

**Comment by Peter226** [comment]
I guess I misinterpreted your phrasing then. Check your Intel Arc driver version with GPU-Z

**Comment by RogerWeihrauch** [comment]
@Peter226 
No prob.
Pls see:
![grafik](https://github.com/user-attachments/assets/c702574f-3b49-4a93-bdf3-614bb21e9170)
In my eyes, it's the latest I could get.
Regards,
Roger

**Comment by Peter226** [comment]
Yeah you should be good then I have one earlier version. The "only for me" is probably not the issue since mine is installed the same: "C:\Users\peter\AppData\Local\Programs\AI Playground"
Unless, do you have foreign characters in your windows username? Python programs have notoriously poor support for foreign characters

**Comment by RogerWeihrauch** [comment]
@Peter226 
Well, so, no so called 'foreign' characters.
just using an en_US environment in Windows.
May be: a complete Windows restart needed? This is the only thing i missed out 'til now.
What u think?
Regards,
Roger

**Comment by Peter226** [comment]
Foreign characters could be áéíóöőüű etc... I know it's not foreign to you and me but in computer land whatever is not in ASCII is sadly still not 100% supported even to this day

About the issue yeah, always restart windows if in doubt

**Comment by RogerWeihrauch** [comment]
@Peter226 
Ok, will try w/ reboot.
Thanks a lot 'til now.
May be I will return to you on this.
Regards,
Roger

**Comment by RogerWeihrauch** [comment]
@Peter226 
Well, no success also w/ this.
hmm, maybe to reinstall it w/ 'for all users' option?
What u think?
Regards,
Roger

**Comment by RogerWeihrauch** [comment]
.. well there was also a int on installing latest VC++ releases.
I did also try its 64bit version; no success.
Regards,
Roger

**Comment by RogerWeihrauch** [comment]
@Peter226 
Also, may be some hints on debugging those?
HowTo?
Regards,
Roger

**Comment by RogerWeihrauch** [comment]
Ok, got it up and running now. (and it also runs in 'only for me' setup)
Here's what I did (and my mistakes), since I got the same error as in:
https://github.com/intel/AI-Playground/issues/57#top

1) error:
- 1st: Installed AI Playground directly after downloading
-  2nd: did not install latest VC++ BEFORE installing complete 'AI PLaygrount' package
2) so, I had to uninstall all ÁI Playground' related files/programs AND do a complete reboot of Windows
3) restarting windows AND installing VC++ (in my case the 64bit version was enough)
4) reboot Windows again
5) install the 'AI Playground'package now
6) shoudl be startable now directly afer installing
7) after some seconds the hint as UI on downloading the 'RunDiffusion/Juggernaught-XL-v9' appeared and I followed instructions
8) after thsi additional package donwload a rendering procedure should start up and
show you after 1-2 min a picture which could now be 'manipulated'
9) have fun and test it out

Regards,
Roger

**Comment by Nuullll** [comment]
Fixed by v1.22
